### DISCUSSION TITLE
tcltk error

### DISCUSSION THREAD

**Discussion by SYOUNG9836** [r-hub/rhub#624] [Q&A] [answered]
![image](https://github.com/r-hub/rhub/assets/78527477/bbbeb378-35b0-4bfd-b5b3-f7cac77a79a9)

I have one more question.

Error related to tcltk:
no DISPLAY variable so Tk is not available

Could this issue also be due to the r code rather than the yaml file?"

This issue does not occur on Windows, but it does happen on Mac and Linux.

**Comment by gaborcsardi** [comment]
I forgot the details, but it is possible that you'll need to work around it on GHA. Can you link to the failing builds?

**Reply by SYOUNG9836** [reply]
Here is the link. https://github.com/SYOUNG9836/rNFI/actions/runs/9576852564

**Reply by gaborcsardi** [reply]
First, you are using some early version of the R-hub workflows. Install/update the rhub package and call 
```r
rhub::rhub_setup(overwrite = TRUE)
```
This probably fixes the macOS builds.

**Reply by SYOUNG9836** [reply]
It worked! Thanks for fixing the macOS builds.

**Comment by gaborcsardi** [comment, marked as answer]
In general the problem is that your package does not work or even load without an X server. On macOS the R-hub actions do install an X server now, so that works. But on Linux that does not happen.

We could try installing an X server on Linux, but if you think that your package should at least load without an X server, you should modify it to import BiodiversityR on demand, and not from the `NAMESPACE` file:

```r
> library(rNFI)
Error: package or namespace load failed for ‘rNFI’:
 .onLoad failed in loadNamespace() for 'Rcmdr', details:
  call: structure(.External(.C_dotTclObjv, objv), class = "tclObj")
  error: [tcl] invalid command name "tk_messageBox".
> library(BiodiversityR)
Error: package or namespace load failed for ‘BiodiversityR’:
 .onLoad failed in loadNamespace() for 'Rcmdr', details:
  call: structure(.External(.C_dotTclObjv, objv), class = "tclObj")
  error: [tcl] invalid command name "tk_messageBox".
> library(Rcmdr)
Error: package or namespace load failed for ‘Rcmdr’:
 .onLoad failed in loadNamespace() for 'Rcmdr', details:
  call: structure(.External(.C_dotTclObjv, objv), class = "tclObj")
  error: [tcl] invalid command name "tk_messageBox".
```

**Reply by SYOUNG9836** [reply]
I’ve resolved it. Thank you so much for your help. I really appreciate your time and effort!
